types/src/block/output/mod.rs

```diff
@@ -184,6 +184,48 @@ impl Output {
         }
     }
 
+    /// Returns the address that is required to unlock this [`Output`] and the alias or nft address that gets
+    /// unlocked by it, if its an alias or nft.
+    pub fn unlocked_addresses(
```

Comment:
(trying to think of a better name)

Reply:
Maybe `required_and_unlocked_address()`?

Reply:
Sounds good to me